packages/router/utils.ts

```diff
@@ -365,24 +375,24 @@ export function convertRoutesToDataRoutes(
     if (isIndexRoute(route)) {
       let indexRoute: AgnosticDataIndexRouteObject = {
         ...route,
-        hasErrorBoundary: detectErrorBoundary(route),
+        ...enhanceAgnosticRoute(route),
         id,
       };
       manifest[id] = indexRoute;
       return indexRoute;
     } else {
       let pathOrLayoutRoute: AgnosticDataNonIndexRouteObject = {
         ...route,
+        ...enhanceAgnosticRoute(route),
```

Comment:
The previous code _always_ set the `hasErrorBoundary` property, but the new `enhanceAgnosticRoute` function isn't guaranteed to return a value for this property. Should we maybe update the `EnhanceAgnosticRouteFunction` type so that it has to return a value for `hasErrorBoundary`?

Reply:
yeah that's probably safest to avoid any mismatches between `ErrorBoundary -> errorElement` and `errorElement -> hasErrorBoundary`

Reply:
Updated in [bd37261](https://github.com/remix-run/react-router/pull/10287/commits/bd37261bc2ed70b841349b1a20177cc3e5f6f225)